File: flowcore/src/main/java/com/raizlabs/android/dbflow/annotation/UniqueGroup.java
package com.raizlabs.android.dbflow.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Description:
 */
@Target(ElementType.ANNOTATION_TYPE)
@Retention(RetentionPolicy.SOURCE)
public @interface UniqueGroup {

    /**
     * @return The number that columns point to to use this group
     */
    int groupNumber();

    /**
     * @return The conflict action that this group takes.
     */
    ConflictAction onUniqueConflict() default ConflictAction.FAIL;
}

File: core/src/commonMain/kotlin/com/dbflow5/annotation/UniqueGroup.kt
package com.dbflow5.annotation

/**
 * Description:
 */
@Target(AnnotationTarget.ANNOTATION_CLASS)
@Retention(AnnotationRetention.SOURCE)
annotation class UniqueGroup(
        /**
         * @return The number that columns point to to use this group
         */
        val groupNumber: Int,
        /**
         * @return The conflict action that this group takes.
         */
        val uniqueConflict: ConflictAction = ConflictAction.FAIL)
